### PR TITLE
fix: Cap django requirement at <4

### PR DESCRIPTION
**Pull request by marksweb**
## Description
Cap the django requirement at less than 4 as we don't support it yet.

## Related resources
* #7179

## Checklist
* [x] I have opened this pull request against ``develop``
* [ ] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 REQUIREMENTS = [
-    'Django>=2.2',
+    'Django>=2.2,<4',
```
Comment (Aiky30) [inline setup.py:9]: @marksweb Does this actually work? 

In the past I've found pip to be very picky with dependancy numbers. For example it doesn't know that 1.0 is 1 when I have tried in the past.

Comment (Aiky30) [inline setup.py:9]:
```suggestion
    'Django>=2.2,<4.0',
```

Comment (marksweb) [reply setup.py:9]: @Aiky30 To confirm this...

```
(.venv) mwalker proj % pip install "django>=3.2,<4"
Requirement already satisfied: django<4,>=3.2 in ./.venv/lib/python3.9/site-packages (3.2.11)
Requirement already satisfied: asgiref<4,>=3.3.2 in ./.venv/lib/python3.9/site-packages (from django<4,>=3.2) (3.4.1)
Requirement already satisfied: sqlparse>=0.2.2 in ./.venv/lib/python3.9/site-packages (from django<4,>=3.2) (0.4.2)
Requirement already satisfied: pytz in ./.venv/lib/python3.9/site-packages (from django<4,>=3.2) (2021.3)
(.venv) mwalker proj % 
```

Comment (vinitkumar) [reply setup.py:9]: @marksweb I agree with @Aiky30 here. 

This is also high wagtail does it. 

https://github.com/wagtail/wagtail/blob/main/setup.py#L23

Comment (crydotsnake) [reply setup.py:9]: Also `'Django>=2.2,<4.0',` looks nicer by reading in my opinion.

Comment (marksweb) [reply setup.py]: Ok chaps, I've changed it 👍 

@Aiky30 @vinitkumar @crydotsnake

**Review by crydotsnake — APPROVED** [review]

**Pull request merged**